test/integration/imagechange_buildtrigger_test.go

```diff
@@ -64,6 +69,9 @@ func TestSimpleImageChangeBuildTrigger(t *testing.T) {
 		t.Fatalf("Expected build with base image %s, got %s", "registry:8080/openshift/test-image:ref-2", newBuild.Parameters.Strategy.DockerStrategy.BaseImage)
 	}
 
+	event = <-watch2.ResultChan()
+	event = <-watch2.ResultChan()
```

Comment:
you sure the issue isn't that we're doing 1 build for the initial buildconfig when the imagerepo points to ref-1, and then a second build (which updates the buildconfig again) when the imgaerepo is pointed to ref-2?